storage/innobase/log/log0recv.cc

```diff
@@ -3627,15 +3633,63 @@ static void recv_recovery_begin(log_t &log, lsn_t *contiguous_lsn,
 
   bool finished = false;
 
+  static int redo_applied = 0;
+  static int redo_applied_total = 0;
+
+
+
   while (!finished) {
     lsn_t end_lsn = start_lsn + RECV_SCAN_SIZE;
 
     recv_read_log_seg(log, log.buf, start_lsn, end_lsn);
 
+    bool flushed = false;
     finished = recv_scan_log_recs(log, max_mem, log.buf, RECV_SCAN_SIZE,
                                   checkpoint_lsn, start_lsn, contiguous_lsn,
-                                  &log.scanned_lsn, to_lsn);
+                                  &log.scanned_lsn, to_lsn, &flushed);
+
+    redo_applied += RECV_SCAN_SIZE;
+    if (flushed) {
+      redo_applied_total += redo_applied;
+
+      auto per = ((double)redo_applied / (double)srv_log_file_size) * 100;
+      auto total_applied_per =
+          ((double)redo_applied_total / (double)srv_log_file_size) * 100;
 
+      ib::info() << "redo applied in last iteration:" << per
+                 << "%\n total redo applied:" << total_applied_per << "%";
+
+      /* find the availabe memory */
+      auto xtrabackup_max_memory =
+          (ulint)(double(sysconf(_SC_AVPHYS_PAGES) * sysconf(_SC_PAGESIZE) *
+                         xtrabackup_free_memory_per) /
+                  100);
+
+      ib::info() << "Maximum memory for buffer pool:" << xtrabackup_max_memory;
+
+      /* gradually increase the buffer pool until it can processed redo log in
+       * one iteration or it hit threshold */
```

Comment:
this style of comment not allowed/ recommened

remove the the * on next line and align to the top comment